qutebrowser/browser/network/networkmanager.py

```diff
@@ -295,6 +297,25 @@ def adopt_download(self, download):
         download.destroyed.connect(self.on_adopted_download_destroyed)
         download.do_retry.connect(self.adopt_download)
 
+    def setReferer(self, req, current_url):
+        """Set the referer header."""
+        referer_header_conf = config.get('network', 'referer-header')
+
+        try:
+            if referer_header_conf == 'never':
+                # Note: using ''.encode('ascii') sends a header with no value,
+                # instead of no header at all
+                req.setRawHeader('Referer'.encode('ascii'), QByteArray())
+            elif (referer_header_conf == 'same-domain' and
+                    not urlutils.same_domain(req.url(), current_url)):
+                req.setRawHeader('Referer'.encode('ascii'), QByteArray())
+            # If refer_header_conf is set to 'always', we leave the header
+            # alone as QtWebKit did set it.
+        except urlutils.InvalidUrlError:
+            # req.url() or current_url can be invalid - this happens on
+            # https://www.playstation.com/ for example.
+            pass
+
```

Comment:
Is it intended that you moved `setReferer` to its own function? Not that I'm opposed to it, but it confused me and I thought it somehow was a change you accidentally got from master or so.

If it is, can you please rename it to `set_referer`? The convention I use for qutebrowser is to use `snake_case` for everything qutebrowser-specific, and `camelCase` for things overriding Qt methods.